File: csharp/2021/day2/day2.cs
using System;
using System.IO;

namespace Advent2021
{
    class Day2
    {
        static void Main(string[] args)
        {
            string[] lines = File.ReadAllLines("./day2.in");

            int aim = 0, x = 0, depth = 0;

            foreach (string line in lines) 
            {
                int magnitude = Int32.Parse(line.Split(" ")[1]);
                switch (line[0]) 
                {
                    case 'f':
                        x += magnitude;
                        depth += aim * magnitude;
                        break;
                    case 'd':
                        aim += magnitude;
                        break;
                    case 'u':
                        aim -= magnitude;
                        break;
                }
            }
            Console.WriteLine($"Part 1: {aim * x}\nPart 2: {x * depth}");
        }
    }
}

File: python/2021/day2/day2.py
#!/usr/bin/env python3
#import solution

with open('./day2.in', 'r') as file:
    data = [(line[0], int(line.split(' ')[1])) for line in file.read().splitlines()]

def part1(data):
    x, depth = 0, 0

    for dir, amount in data:
        if dir == 'f':
            x += amount
        elif dir == 'd':
            depth += amount
        else: # u
            depth -= amount

    print('Part 1:', x * depth)

def part2(data):
    x, depth, aim = 0, 0, 0

    for dir, amount in data:
        if dir == 'f':
            x += amount
            depth += aim * amount
        elif dir == 'd':
            aim += amount
        else: # u
            aim -= amount

    print('Part 2:', x * depth)

part1(data)
part2(data)